File: signature/jquery_migrate.go
package signature

var JQueryMigrateSignature = Signature{
	Name:        "jQuery Migrate",
	Description: "jQuery Migrate makes older code compatible with jQuery 3.0+.",
	Cpe:         "cpe:/a:jquery_migrate_project:jquery_migrate",

	DetectPattern: DetectPattern{
		Urls: []string{
			"jquery-migrate",
		},
		Bodies: []string{
			"jQuery Migrate",
			"jQuery.migrate",
		},
	},
	VersionPattern: VersionPattern{
		Urls: []string{
			`jquery-migrate\.min\.js\?ver=(\d+\.\d+\.\d+)`,
		},
		Bodies: []string{
			`jQuery Migrate v(\d+\.\d+\.\d+)`,
		},
	},
}

File: src/signatures/technologies/jquery_migrate.ts
import type { Signature } from "../_types.js";
import { jquerySignature } from "./jquery.js";

export const jqueryMigrateSignature: Signature = {
  name: "jQuery Migrate",
  description:
    "Query Migrate is a javascript library that allows you to preserve the compatibility of your jQuery code developed for versions of jQuery older than 1.9.",
  rule: {
    confidence: "high",
    urls: [
      "jquery[.-]?migrate[./-]?(\\d+\\.\\d+\\.\\d+)?",
      "jquery[.-]?migrate.*?ver=(\\d+\\.\\d+\\.\\d+)?",
    ],
    javascriptVariables: {
      "jQuery.migrateVersion": "(\\d+\\.\\d+\\.\\d+)?",
      "jQuery.migrateWarnings": "",
      jqueryMigrate: "",
    },
  },
  impliedSoftwares: [jquerySignature.name],
};
